Compilation failures with gasnet locally

I'm seeing the following error while trying to do a gasnet build of the server with master:
```
/Users/pierce/proj/arkouda//src/MultiTypeSymEntry.chpl:317: In function 'createSymEntry':
/Users/pierce/proj/arkouda//src/MultiTypeSymEntry.chpl:318: error: ambiguous call 'makeDistArray([domain(1,int(64),one)] int(64))'
/Users/pierce/proj/arkouda//src/compat/e-132/SymArrayDmapCompat.chpl:77: note: candidates are: makeDistArray(a: [])
/Users/pierce/proj/arkouda//src/compat/e-132/SymArrayDmapCompat.chpl:84: note:                 makeDistArray(a: [])
  /Users/pierce/proj/arkouda//src/KExtremeMsg.chpl:52: called as createSymEntry(a: [domain(1,int(64),one)] int(64), max_bits: int(64))
note: generic instantiations are underlined in the above callstack
make: *** [arkouda_server] Error 1
```
@jeremiah-corrado is there any chance this is related to the recent updates in `SymArrayDmapCompat`?

--- a/src/compat/e-132/SymArrayDmapCompat.chpl
+++ b/src/compat/e-132/SymArrayDmapCompat.chpl
@@ -82,7 +82,7 @@ module SymArrayDmapCompat
     }
 
     proc makeDistArray(in a: [?D] ?etype) throws
-      where D.rank == 1
+      where D.rank == 1 && (MyDmap == Dmap.defaultRectangular || !a.isDefaultRectangular())
     {
       var res = D.tryCreateArray(etype);
       res = a;
@@ -90,7 +90,7 @@ module SymArrayDmapCompat
     }
 
     proc makeDistArray(in a: [?D] ?etype) throws
-      where D.rank > 1
+      where D.rank > 1 && (MyDmap == Dmap.defaultRectangular || !a.isDefaultRectangular())
     {
       return a;
     }

--- a/src/compat/gt-132/SymArrayDmapCompat.chpl
+++ b/src/compat/gt-132/SymArrayDmapCompat.chpl
@@ -82,7 +82,7 @@ module SymArrayDmapCompat
     }
 
     proc makeDistArray(in a: [?D] ?etype) throws
-      where D.rank == 1
+      where D.rank == 1 && (MyDmap == Dmap.defaultRectangular || !a.isDefaultRectangular())
     {
       var res = D.tryCreateArray(etype);
       res = a;
@@ -90,7 +90,7 @@ module SymArrayDmapCompat
     }
 
     proc makeDistArray(in a: [?D] ?etype) throws
-      where D.rank > 1
+      where D.rank > 1 && (MyDmap == Dmap.defaultRectangular || !a.isDefaultRectangular())
     {
       return a;
     }

